File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        best = 0
        dict = {}
        for n in nums:
            if n not in dict:
                left = 0 if n - 1 not in dict else dict[n - 1]
                right = 0 if n + 1 not in dict else dict[n + 1]
                length = left + right + 1
                dict[n] = length
                dict[n - left] = length
                dict[n + right] = length
                best = max(best, length)
        return best

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Integer> lengthMap = new HashMap<>();
        
        int max = 0;
        for (int n : nums) {
            if (!lengthMap.containsKey(n)) {
                int left = (lengthMap.containsKey(n - 1)) ? lengthMap.get(n - 1) : 0;
                int right = (lengthMap.containsKey(n + 1)) ? lengthMap.get(n + 1) : 0;
                int length = left + right + 1;
                lengthMap.put(n, length);
                lengthMap.put(n - left, length);
                lengthMap.put(n + right, length);
                max = (max < length) ? length : max;
            }
        }

        return max;
    }
}